modules/Activities/activities_attendance.php

```diff
@@ -73,40 +75,27 @@
-        $studentResult = $connection2->prepare($sql);
-        $studentResult->execute($data);
-
-
-        $data = array('gibbonActivityID' => $gibbonActivityID);
-        $sql = "SELECT gibbonSchoolYearTermIDList, maxParticipants, programStart, programEnd, (SELECT COUNT(*) FROM gibbonActivityStudent JOIN gibbonPerson ON (gibbonActivityStudent.gibbonPersonID=gibbonPerson.gibbonPersonID) WHERE gibbonActivityStudent.gibbonActivityID=gibbonActivity.gibbonActivityID AND gibbonActivityStudent.status='Waiting List' AND gibbonPerson.status='Full') AS waiting FROM gibbonActivity WHERE gibbonActivityID=:gibbonActivityID";
-        $activityResult = $connection2->prepare($sql);
-        $activityResult->execute($data);
+        $activityResult = $container->get(ActivityStudentGateway::class)->selectActivityByStudents($gibbonActivityID);
 
     if ($studentResult->rowCount() < 1 || $activityResult->rowCount() < 1) {
         echo $page->getBlankSlate();
 
         return;
     }
 
-
     $students = $studentResult->fetchAll();
 
-    $data = array('gibbonActivityID' => $gibbonActivityID);
-    $sql = 'SELECT gibbonActivityAttendance.date, gibbonActivityAttendance.timestampTaken, gibbonActivityAttendance.attendance, gibbonPerson.preferredName, gibbonPerson.surname FROM gibbonActivityAttendance, gibbonPerson WHERE gibbonActivityAttendance.gibbonPersonIDTaker=gibbonPerson.gibbonPersonID AND gibbonActivityAttendance.gibbonActivityID=:gibbonActivityID';
-    $attendanceResult = $connection2->prepare($sql);
-    $attendanceResult->execute($data);
+    $attendanceResult = $container->get(ActivityAttendanceGateway::class)->selectStudentAttendanceByActivity($gibbonActivityID);
 
     // Gather the existing attendance data (by date and not index, should the time slots change)
     $sessionAttendanceData = array();
 
     while ($attendance = $attendanceResult->fetch()) {
         $sessionAttendanceData[ $attendance['date'] ] = array(
             'data' => (!empty($attendance['attendance'])) ? unserialize($attendance['attendance']) : array(),
-            'info' => sprintf(__('Recorded at %1$s on %2$s by %3$s.'), substr($attendance['timestampTaken'], 11), Format::date(substr($attendance['timestampTaken'], 0, 10)), Format::name('', $attendance['preferredName'], $attendance['surname'], 'Staff', false, true)),
+            'info' => __('Recorded at %1$s on %2$s by %3$s.'), substr($attendance['timestampTaken'], 11), Format::date(substr($attendance['timestampTaken'], 0, 10)), Format::name('', $attendance['preferredName'], $attendance['surname'], 'Staff', false, true),
```

Comment:
It looks like this line has removed the sprintf, but not updated the syntax, so this string will likely cause an error. In this case, lets restore the sprintf and refactor strings like this in a future PR.

Reply:
noted and add the sprintf back